modules/node-shared/src/main/scala/org/tessellation/node/shared/domain/statechannel/FeeCalculator.scala

```diff
@@ -0,0 +1,89 @@
+package org.tessellation.node.shared.domain.statechannel
+
+import java.math.{MathContext, RoundingMode}
+
+import cats.MonadThrow
+import cats.data.{NonEmptyList, NonEmptySet}
+import cats.syntax.all._
+
+import scala.collection.immutable.SortedMap
+
+import org.tessellation.currency.schema.currency.SnapshotFee
+import org.tessellation.schema.SnapshotOrdinal
+import org.tessellation.schema.balance.Balance
+
+import eu.timepit.refined.api.Refined
+import eu.timepit.refined.auto._
+import eu.timepit.refined.numeric.Interval
+import eu.timepit.refined.types.numeric.{NonNegDouble, NonNegInt, NonNegLong}
+
+case class FeeCalculatorConfig(baseFee: NonNegLong, stakingWeight: NonNegDouble, computationalCost: NonNegLong, proWeight: NonNegDouble)
+
+object FeeCalculatorConfig {
+  val noFee: FeeCalculatorConfig = FeeCalculatorConfig(baseFee = 0L, stakingWeight = 0.0, computationalCost = 1L, proWeight = 0.0)
+}
+
+sealed trait FeeCalculator[F[_]] {
+  def calculateRecommendedFee(
+    maybeOrdinal: Option[SnapshotOrdinal],
+    delay: NonNegLong = 0L
+  )(staked: Balance, sizeKb: NonNegInt, tip: NonNegLong = 0L, proScore: FeeCalculator.ProScore = 0.0d): F[SnapshotFee]
+}
+
+object FeeCalculator {
+  def make[F[_]: MonadThrow](configByOrdinal: SortedMap[SnapshotOrdinal, FeeCalculatorConfig]): FeeCalculator[F] =
+    new FeeCalculator[F] {
+      private val allConfigs: NonEmptyList[FeeCalculatorConfig] =
+        NonEmptyList
+          .fromList(configByOrdinal.values.toList)
+          .getOrElse(NonEmptyList.one(FeeCalculatorConfig.noFee))
+
+      private def getConfigByOrdinal(ordinal: SnapshotOrdinal): FeeCalculatorConfig =
+        configByOrdinal
+          .rangeTo(ordinal)
+          .lastOption
+          .map { case (_, config) => config }
+          .getOrElse(FeeCalculatorConfig.noFee)
+
+      /** Calculates the recommended fee for given stake, size and pro score and adds optional tip on top. Calculates the worst case
+        * scenario (highest) fee based on the provided configs range picked as follows:
+        *   - all configs when ordinal is None
+        *   - ordinal to ordinal + delay when ordinal is non empty.
+        */
```

Comment:
Do we want such scaladoc explaining a bit, or do we consider a clean code enough?

Reply:
If you aimed to write a scaladoc, it should be sth like that:
```scala
/** Calculates the recommended fee for a given stake, size, pro score, and optional tip.
  *
  * This method calculates the worst-case scenario (highest) fee based on the provided configs range.
  * The range is determined as follows:
  *   - If `ordinal` is `None`, all configs are considered.
  *   - If `ordinal` is non-empty, the range is from `ordinal` to `ordinal + delay`.
  *
  * @param maybeOrdinal an optional [[SnapshotOrdinal]] representing the starting ordinal for the config range
  * @param delay a [[NonNegLong]] representing the number of ordinals to include in the config range
  * @param staked the [[Balance]] of the staked amount
  * @param sizeKb the size of the transaction in kilobytes as a [[NonNegInt]]
  * @param tip an optional [[NonNegLong]] representing the additional tip amount to include in the fee calculation
  * @param proScore the [[ProScore]] of the transaction sender
  * @return an [[F]] containing the calculated [[SnapshotFee]] representing the recommended fee
  */
```

Reply:
I didn't aim for a fully fledged scaladoc. I wanted to give enough insight, so that it's easier to understand the reasoning. My question was rather about whether we want such minimalistic description or do we consider code to be enough (self explaining I guess). We don't add much descriptions like that in general.

Reply:
If you want to add some side note, go for it. but IMO it doesn't add much value since the code is self explanatory. Anyway, take a look at my rewrite of the comment which in my opinion is more straightforward to understand.